### ISSUE TITLE
Possible regression on variables defined as props

### ISSUE THREAD

**Issue by meriadec** [yannickcr/eslint-plugin-react#2330]
Not sure it's the right place to open the issue because it's also kinda related to [eslint-plugin-flowtype](https://github.com/gajus/eslint-plugin-flowtype). After bisecting it seems that the `react/prop-types` rule has a regression introduced since https://github.com/yannickcr/eslint-plugin-react/pull/2301 when used with Flowtype.

example better than a long explanation:

```js
// @flow

import React from "react";

const foo = {
  some: "data",
};

type Props = {
  derived: { [$Keys<typeof foo>]: string },
};

export function Bar(props: Props) {
  const { derived } = props;
  return <div className={derived.some}>foo</div>;
}
```

**before https://github.com/yannickcr/eslint-plugin-react/pull/2301**: `No errors`
**after https://github.com/yannickcr/eslint-plugin-react/pull/2301**: 

```
15:34  error  'derived.some' is missing in props validation  react/prop-types
```

@golopot does it rings a bell on your side? Would totally understand if the issue is irrelevant and is not in the scope of eslint-plugin-react.

Have a good day

**Comment by golopot** [comment]
The problem seems to that this plugin do not recognize "indexer property" in flow types.

```ts
// This will pass
type Props = { a: { b: string } };

export function Bar(props: Props) {
  return <>{props.a.b}</>;
}

// This will warn
type Props = { a: { [string]: string } };

export function Bar(props: Props) {
  return <>{props.a.b}</>;
                    ~ // 'a.b' is missing in props validationeslint(react/prop-types)
}


```
